lib/matplotlib/pyplot.py

```diff
@@ -1146,6 +1159,23 @@ def subplots(nrows=1, ncols=1, sharex=False, sharey=False, squeeze=True,
                        gridspec_kw=gridspec_kw)
     return fig, axs
 
+def subplots_iterator(nrows=1, ncols=1, show_in_between=False, sharex=False,
+                      sharey=False, squeeze=True, subplot_kw=None,
+                      gridspec_kw=None, **fig_kw):
+    """ Iteratively yields the axis object of a rows x cols subplot and creates new subplots when needed"""
+    for axis in (
```

Comment:
We have both an `Axes` class and and `Axis` class, please be very careful with keeping things consistent (these are `Axes` objects).